store.go

```diff
@@ -1118,9 +1116,18 @@ func (s *store) getROLayerStoresLocked() ([]roLayerStore, error) {
 	if err := os.MkdirAll(rlpath, 0o700); err != nil {
 		return nil, err
 	}
-	for _, store := range s.graphDriver.AdditionalImageStores() {
+
+	stores := s.graphDriver.AdditionalImageStores()
+	if s.imageStoreDir != "" {
+		stores = append([]string{s.graphRoot}, stores...)
+	}
+
+	for _, store := range stores {
 		glpath := filepath.Join(store, driverPrefix+"layers")
-		rls, err := newROLayerStore(rlpath, glpath, s.graphDriver)
+
+		ro := store != s.imageStoreDir && store != s.graphRoot
+
+		rls, err := s.newLayerStore(rlpath, glpath, s.graphDriver, false, ro)
```

Comment:
this is needed because we have this weird specification that a layer must be deleted from the current store if it is found there.  We even have a test for it :/

https://github.com/containers/storage/blob/main/tests/split-store.bats#L96-L101

I am fine if we decide to break this specification, but let's do it separately after we verify with CRI-O folks that it is not necessary.

Reply:
Fine… but I can’t see how that works. `layersToRemove` is only applied to one `rlstore`. That uses `getLayerStoreLocked`, and AFAICS never gets to write to this special member of `roLayerStoresUseGetters`.

I’m probably being stupid.